modules/packages/OrderedSet/Treap.chpl

```diff
@@ -0,0 +1,872 @@
+      `iterable`. If an element from `iterable` is already contained in this
+      orderedSet, it will not be added again. The formal `iterable` must be a type
+      with an iterator named "these" defined for it.
+
+      :arg iterable: A collection of elements to add to this orderedSet.
+      :arg parSafe: If `true`, this orderedSet will use parallel safe operations.
+      :arg comparator: The comparator used to compare elements.
+    */
+    proc init(type eltType, iterable, param parSafe = false, comparator: record = defaultComparator)
+    where canResolveMethod(iterable, "these") lifetime this < iterable {
+      _checkType(eltType); 
+
+      this.eltType = eltType;
+      this.parSafe = parSafe;
+      this.comparator = comparator;
+      this.complete();
+
+      for elem in iterable do _add(elem);
+    }
+
+    /*
+      Inorder traversal for output
+    */
+    pragma "no doc"
+    proc const _lmrVisit(node: nodeType, ch: channel) throws {
+      if node == nil {
+        return;
+      }
+      else {
+        node!.sanityChecks();
+        _lmrVisit(node!.children[0], ch);
+        ch.write(node!.element, ' ');
+        _lmrVisit(node!.children[1], ch);
+      }
+    }
+
+    /*
+      Visit and output elements in order
+    */
+    pragma "no doc"
+    proc const _visit(ch: channel) throws {
+      ch.write('[ ');
+      _lmrVisit(_root, ch);
+      ch.write(']');
+    }
+
+    pragma "no doc"
+    proc ref _add(in x: eltType) lifetime this < x {
+      _insert(_root, x, nil);
+    }
+
+    /*
+      Add a copy of the element `x` to this orderedSet. Does nothing if this orderedSet
+      already contains an element equal to the value of `x`.
+
+      :arg x: The element to add to this orderedSet.
+    */
+    proc ref add(in x: eltType) lifetime this < x {
+
+      // Remove `on this` block because it prevents copy elision of `x` when
+      // passed to `_add`. See #15808.
+      _enter();
+      _add(x);
+      _leave();
+    }
+
+    proc const _contains(const ref x: eltType): bool {
+      return _find(_root, x) != nil;
+    }
+
+    /*
+      Returns `true` if the given element is a member of this orderedSet, and `false`
+      otherwise.
+
+      :arg x: The element to test for membership.
+      :return: Whether or not the given element is a member of this orderedSet.
+      :rtype: `bool`
+    */
+    proc const contains(const ref x: eltType): bool {
+      var result = false;
+
+      on this {
+        _enter(); 
+        result = _contains(x);
+        _leave();
+      }
+
+      return result;
+    }
+
+    /*
+      Helper procedure to make x become a child of y, in position pos
+    */
+    pragma "no doc"
+    proc _link(x: nodeType, y: nodeType, pos: int) {
+      if x != nil then
+        x!.parent = y;
+      if y != nil then
+        y!.children[pos] = x;
+    }
+
+    /*
+      The rotation will make the node.children[pos] becomes the new root
```

Comment:
I think it's more that it'll make that child take the place of `node`, or `become the new root of the subtree formerly rooted at node`, which is a bit of a mouthful.

Also, these sentences should have a `.` at the end.